clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -72,13 +76,24 @@
     private static final String ISR_KEY_NAME = "isr";
 
     private final Collection<Node> brokers;
+    private final Node controller;
     private final List<TopicMetadata> topicMetadata;
 
+    /**
+     * Constructor for the latest version
+     */
+    public MetadataResponse(List<Node> brokers, int controllerId, List<TopicMetadata> topicMetadata) {
```

Comment:
@gwenshap I think it actually ends up being a bit of "crying wolf" since the brokers would be required to use the old constructors to support the old versions. We would throw, and likely suppress, the deprecation warnings in that case. I think unit tests are the best way to verify this behavior. I will work on increasing the version coverage on this patch and consider updating other apis at a later time.